linode/manager
# feat: [M3-7645] – Support VPC IPv4 Ranges in Add/Edit Linode Config dialog

## Description 📝
> [!NOTE] 
> This branches off from https://github.com/linode/manager/pull/10116 -- the diff should be reduced down to a handful of files once the preceding merges have occurred.

Add support for VPC IPv4 ranges in add/edit linode config dialog

- "Add additional IPv4 ranges" section added in VPC panel
- Logic to include ip_ranges in creation payload

## Changes  🔄
- 

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-01-28 at 10 06 22 PM](https://github.com/linode/manager/assets/125309814/82d18e2c-02c0-4d7b-82fb-37a2518c63da) | ![Screenshot 2024-01-28 at 8 33 21 PM](https://github.com/linode/manager/assets/125309814/a293d097-f282-4f51-844c-890e722ca998) |

## How to test 🧪

### Verification Steps
With the vpc-extra-beta tag on your account, enter the Linode Create flow and confirm:

- The "Add additional IPv4 ranges" section appears in the VPC panel after a subnet is selected
- You can add and remove IP input fields using the "Add IPv4 Range" and the "X" buttons, respectively
- When you click "Add Configuration," the POST request contains the IP ranges you specified in the form in the ip_ranges property
- Once the config is created, go to the detail page of the VPC you assigned the linode to. Expand the inner subnet table and observe the "VPC IPv4 Ranges" column is populated with what you provided (or "None" if you didn't provide any).

## As an Author I have considered 🤔
